lib/SILOptimizer/Utils/OwnershipOptUtils.cpp

```diff
@@ -193,23 +223,21 @@ static bool canFixUpOwnershipForRAUW(SILValue oldValue, SILValue newValue) {
   if (m->getStage() == SILStage::Raw)
     return false;
 
-  // If our old ownership kind is ValueOwnershipKind::None and our new kind is
-  // not, we may need to do more work that has not been implemented yet. So
-  // bail.
-  //
-  // Due to our requirement that types line up, this can only occur given a
-  // non-trivial typed value with None ownership. This can only happen when
-  // oldValue is a trivial payloaded or no-payload non-trivially typed
-  // enum. That doesn't occur that often so we just bail on it today until we
-  // implement this functionality.
-  auto oldOwnershipKind = SILValue(oldValue).getOwnershipKind();
-  if (oldOwnershipKind != OwnershipKind::None)
-    return true;
+  return true;
+}
 
-  // Ok, we have an old ownership kind that is OwnershipKind::None and a new
-  // ownership kind that is not OwnershipKind::None. In that case, for now, do
-  // not perform this transform.
-  return false;
+static bool canFixUpOwnershipForRAUW(SILValue oldValue, SILValue newValue,
```

Comment:
Can you add a comment here similar to the comment on hasValidRAUWOwnership explaining that you also check for transitive uses?

Reply:
oops that comment got dropped. fixed it